### PR TITLE
modify the eddy diffusivity for heat at the top of the canopy

### PR DESCRIPTION
**Pull request by HelinWei-NOAA**
Revised sensible heat flux for under-canopy sublayer in Noah-MP land surface model.
-- Modify the eddy diffusivity coefficients for momentum and heat at the top of the canopy.
-- Tune the parameters cd (the drag coefficient of leaves) and lm (mean mixing length).

### CODE REVIEW

**Comment by barlage** [comment]
@wzzheng90 do you plan additional commits for this PR?

File: physics/module_sf_noahmplsm.f90 [hunk 1]
```diff
@@ -4652,29 +4654,36 @@ subroutine ragrb(parameters,iter   ,vai    ,rhoair ,hg     ,tah    , & !in
...
         tmp1 = vkc * (grav/tah) * hg/(rhoair*cpair)
         if (abs(tmp1) .le. mpe) tmp1 = mpe
         molg = -1. * fv**3 / tmp1
         mozg = min( (zpd-z0mg)/molg, 1.)
+        mozgh = min( (hcan - zpd)/molg, 1.)
        end if
 
        if (mozg < 0.) then
           fhgnew  = (1. - 15.*mozg)**(-0.25)
+            fhgnewh  = 0.74 * (1. - 9.*mozg)**(-0.5)    ! PHIh
```
Comment (barlage) [inline physics/module_sf_noahmplsm.f90:4675]: can you remove this leading space so they line up?

File: physics/module_sf_noahmplsm.f90 [hunk 2]
```diff
@@ -4686,7 +4695,7 @@ subroutine ragrb(parameters,iter   ,vai    ,rhoair ,hg     ,tah    , & !in
 
 ! aerodynamic resistances raw and rah between heights zpd+z0h and z0hg.
 
-       kh  = max ( vkc*fv*(hcan-zpd), mpe )
+       kh  = max ( vkc*fv*(hcan-zpd)/fhgh, mpe )
```
Comment (barlage) [inline physics/module_sf_noahmplsm.f90:4698]: do we need protection against divide by zero here?

**Comment by wzzheng90** [comment]
I modified cwpvt LUT and modified "cd/lm" parameter (reduced half).

File: physics/noahmp_tables.f90 [hunk 3]
```diff
@@ -510,11 +510,11 @@ module noahmp_tables
...
     real :: cwpvt_table(mvt)       !< empirical canopy wind parameter
-      data ( cwpvt_table (i),i=1,mvt) / 0.18, 0.67, 0.18, 0.67, 0.29, 1.00,  &
-     &                            2.00, 1.30, 1.00, 5.00, 1.17, 1.67,        &
-     &                            1.67, 1.67, 0.18, 0.18, 0.18, 0.67,        &
-     &                            1.00, 0.18, 0.00, 0.00, 0.00, 0.00,        &
-     &                            0.00, 0.00, 0.00, 0.00, 0.00, 0.00 /
+      data ( cwpvt_table (i),i=1,mvt) / 0.09, 0.335, 0.09, 0.335, 0.145, 0.50, &
+     &                            1.00, 0.65,  0.50, 2.50, 0.585, 0.835,       &
+     &                            0.835, 0.835, 0.09, 0.09, 0.09, 0.335,       &
+     &                            0.835, 0.835, 0.09, 0.09, 0.09, 0.335,       &
```
Comment (HelinWei-NOAA) [inline physics/noahmp_tables.f90:516]: should be 0.5, 0.09, 0.00, 0.00, 0.00, 0.00?

Comment (HelinWei-NOAA) [reply physics/noahmp_tables.f90:516]: line 516

Comment (wzzheng90) [reply physics/noahmp_tables.f90]: Yes,  Thank you!

**Comment by HelinWei-NOAA** [comment]
@wzzheng90 As you mentioned, the model is very sensitive to the parameter cwpvt. With the latest change, the model didn't complete the case of 20120701. The other cases look fine. My run directory is /scratch1/NCEPDEV/stmp2/Helin.Wei/Prototype8.lsmz/COMROOT

The job log file for the failed case is /scratch1/NCEPDEV/stmp2/Helin.Wei/Prototype8.lsmz/COMROOT/R_20120701/logs/2012070100/gfsfcst.log.0

Any suggestions?

File: physics/module_sf_noahmplsm.f90 [hunk 4]
```diff
@@ -4652,29 +4654,36 @@ subroutine ragrb(parameters,iter   ,vai    ,rhoair ,hg     ,tah    , & !in
...
 
        if (iter == 1) then
           fhg = fhgnew
+          fhgh = fhgnewh
        else
           fhg = 0.5 * (fhg+fhgnew)
+          fhgh = 0.5 * (fhgh+fhgnewh)
        endif
 
        cwpc = (cwp * vai * hcan * fhg)**0.5
```
Comment (barlage) [inline physics/module_sf_noahmplsm.f90:4689]: I suggest we put in some limits like this: cwpc = max(min(cwpc,5.0),1.0)

**Comment by HelinWei-NOAA** [comment]
With that limitation, the model went through that breaking point. The
subset results of this test  can be
found /gpfs/dell3/ptmp/Helin.Wei/Prototype8.lsmf/COMROOT/P8lsmz.sub. The
verification page is https://www.emc.ncep.noaa.gov/mmb/gcp/gfs/p8alsmz. The
surface cape has been increased furthermore.

**Comment by wzzheng90** [comment]
Sounds great!   My test with P8a (case: 20120701) also went through, but without the limitation (cwpc = max(min(cwpc,5.0),1.0)). So, it seems that this modification together with other land changes would result in some other issues.

**Comment by barlage** [comment]
@HelinWei-NOAA are you planning to put this change in the branch? I'm running the spin up now but just realized I am using the branch directly and haven't added this limit.

**Pull request merged**